nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -450,8 +450,9 @@ sub addEntry {
 
     # Include second initrd with secrets
     if (-e -x "$path/append-initrd-secrets") {
-        my $initrdName = basename($initrd);
-        my $initrdSecretsPath = "$bootPath/kernels/$initrdName-secrets";
+        # Name the initrd secrets after the system from which they're derived.
+        my $systemName = basename(Cwd::abs_path("$path"));
+        my $initrdSecretsPath = "$bootPath/kernels/$systemName-secrets";
```

Comment:
This makes sense a lot of sense: it should have always been like this. 
I don't know why there's a `$initrdName` if the name is going to be `initrd` anyway... maybe it's not intentional.